src/Microsoft.Health.Fhir.Shared.Core/Features/Operations/Export/GroupMemberExtractor.cs

```diff
@@ -96,7 +96,7 @@ private async Task<HashSet<string>> GetGroupPatientIdsHelper(string groupId, Dat
 
             if (groupResource == null)
             {
-                throw new ResourceNotFoundException($"Group {groupId} was not found.");
+                throw new ResourceNotFoundException($"Group {groupId} was not found.", true);
```

Comment:
Raise GroupNotFoundException?

ResourceNotFoundException is a generic exception today. If we add a new parameter called GroupNotFound, then it'll make the parameter useless for most of all its usages.

Reply:
We had long conversation and changed plan altogether.